model/src/main/kotlin/DependencyGraph.kt

```diff
@@ -61,6 +61,30 @@ data class DependencyGraph(
      */
     val scopes: Map<String, List<RootDependencyIndex>>
 ) {
+    companion object {
+        /**
+         * Return a name for the given [scope][scopeName] that is qualified with parts of the identifier of the given
+         * [project]. When constructing a dependency graph from multiple projects, it must be ensured that the scope
+         * names are unique. To achieve this, they are qualified with the project ID.
+         */
+        fun qualifyScope(project: Project, scopeName: String): String =
+            qualifyScope(project.id, scopeName)
+
+        /**
+         * Return a name for the given [scope][scopeName] that is qualified with parts of the given [projectId]. When
+         * constructing a dependency graph from multiple projects, it must be ensured that the scope names are unique.
+         * To achieve this, they are qualified with the project ID.
+         */
+        fun qualifyScope(projectId: Identifier, scopeName: String): String =
+            "${projectId.namespace}:${projectId.name}:${projectId.version}:$scopeName"
+
+        /**
+         * Extract the plain (un-qualified) scope name from the given qualified [scopeName]. If the passed in
+         * [scopeName] is not qualified, return it unchanged.
+         */
+        fun unQualifyScope(scopeName: String): String = scopeName.substringAfterLast(':')
```

Comment:
It should be as we have

https://github.com/oss-review-toolkit/ort/blob/274d7b2273ba9601c52ece0d7b3298e93c233ad1/model/src/main/kotlin/Identifier.kt#L83-L86

or?

Reply:
Yes, I was using `Identifier` as a model and was hoping the same.

Reply:
The code above doesn't help because the scope is not part of the identifier. But I think the risk pretty low that scope names contain ":". But anyway, you could change that to return the string after the third ":" to be on the safe side.

Reply:
The function has been renamed and now returns the part after the 3rd separator.